lib/modules/manager/argocd/extract.ts

```diff
@@ -88,15 +107,24 @@ function processAppSpec(
       ? definition.spec
       : definition.spec.template.spec;
 
-  const deps: (PackageDependency | null)[] = [];
+  let deps: PackageDependency[] = [];
 
   if (is.nonEmptyObject(spec.source)) {
-    deps.push(processSource(spec.source));
+    deps = processSource(spec.source);
   }
 
   for (const source of coerceArray(spec.sources)) {
-    deps.push(processSource(source));
+    deps = deps.concat(processSource(source));
+  }
+
+  return deps;
+}
+
+function processKustomizeImage(image: string): PackageDependency | null {
+  const parts = kustomizeImageRe.exec(image);
```

Comment:
I'm not quite happy with having both the parameter to the function, as well as the capture group named `image`, even though both are accurate, as the former is an item from `images` list, the latter is - well - an image.

Do you have a suggestion for a better name of either of the two?

Reply:
see above